verified-broadcast/assets/setup.private.js

```diff
@@ -0,0 +1,164 @@
+async function updateEnvironmentVariable(twilioClient, context, key, value) {
+  if (!context.SERVICE_SID && context.ENVIRONMENT_SID) {
+    context[key] = value;
+    process.env[key] = value;
+    return;
+  }
+  if (context[key] === value) {
+    // variable already exists
+  } else if (context[key]) {
+    const existingVariables = await twilioClient.serverless
+      .services(context.SERVICE_SID)
+      .environments(context.ENVIRONMENT_SID)
+      .variables.list();
+    const [variableToUpdate] = existingVariables.filter(
+      (variable) => variable.key === key
+    );
+    variableToUpdate.update({ value });
+    context[key] = value;
+    process.env[key] = value;
+  } else {
+    await twilioClient.serverless
+      .services(context.SERVICE_SID)
+      .environments(context.ENVIRONMENT_SID)
+      .variables.create({
+        key,
+        value,
+      });
+    context[key] = value;
+    process.env[key] = value;
+  }
+}
+
+async function getMessagingServiceFromPhoneNumber(
+  twilioClient,
+  targetPhoneNumber
+) {
+  const messagingServices = await twilioClient.messaging.services.list({
+    limit: 2000,
+  });
+
+  const phoneNumberMap = new Map();
+  const requests = messagingServices.map(async (service) => {
+    const numbers = await service.phoneNumbers.list({ limit: 2000 });
+    numbers.forEach(({ phoneNumber }) => {
+      phoneNumberMap.set(phoneNumber, service.sid);
+    });
+    return true;
+  });
+  await Promise.all(requests);
+
+  if (phoneNumberMap.has(targetPhoneNumber)) {
+    return phoneNumberMap.get(targetPhoneNumber);
+  }
+  return null;
+}
+
+async function setupMessagingService(twilioClient, context) {
+  if (!context.TWILIO_PHONE_NUMBER) {
+    throw new Error(
+      'Cannot create a Messaging Service without a TWILIO_PHONE_NUMBER'
+    );
+  }
+
+  const existingServiceSid = await getMessagingServiceFromPhoneNumber(
```

Comment:
Is the idea here that we want to use the messaging service if one is already set up for the phone number? If that's the case, could we expect the customer to provide the SID at setup instead of going through this step to find it?

Reply:
I'm trying to reduce the amount of setup that people have to do in the Twilio Console to zero. Since Quick Deploy doesn't support picking anything other than phone numbers from a drop-down today we'd have to have people go and create their own messaging service at which point they could also create the other services.